Scripts/Event/RestrictableSingleEventProxyEmitter.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace VRTK.Core.Event
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using VRTK.Core.Extension;
+    using VRTK.Core.Rule;
+
+    public abstract class RestrictableSingleEventProxyEmitter<TValue, TEvent> : SingleEventProxyEmitter<TValue, TEvent> where TEvent : UnityEvent<TValue>, new()
+    {
+        /// <summary>
+        /// Determines whether the received payload is valid to be re-emitted.
+        /// </summary>
+        [Tooltip("Determines whether the received payload is valid to be re-emitted.")]
+        public RuleContainer receiveValidity;
+
+        /// <summary>
+        /// Gets the target for the validity check.
+        /// </summary>
+        /// <returns>The target to check on.</returns>
+        protected abstract object GetAcceptsTarget();

```

Comment:
`GetPotentialTarget` or `GetTargetToCheck` instead perhaps? Kinda ready wrong with the `Accepts` in there as it talks about the API's name it's used for rather than describing the returned object.